server/filestore.go

```diff
@@ -4491,7 +4495,7 @@ func (fs *fileStore) removeMsg(seq uint64, secure, viaLimits, needFSLock bool) (
 
 	// If we are tracking multiple subjects here make sure we update that accounting.
 	mb.removeSeqPerSubject(sm.subj, seq)
-	fs.removePerSubject(sm.subj)
+	wasLast := fs.removePerSubject(sm.subj)
```

Comment:
Since removePerSubject is fs scoped can just do this inside that function IMO..

Reply:
I did try that, but there's a couple cases where it doesn't make sense, i.e. in `expireMsgs` and `expireMsgsOnRecover`, we want to first make sure that the deleted message isn't itself a delete marker (so that we don't end up making more of them endlessly), but we still need to do `removePerSubject` regardless.

Reply:
But that can all be in removePerSubject() if we pass in bool yes that indicates if this is a marker?

Reply:
Have updated it so we pass in a bool on whether or not to queue a marker.